packages/frontend/src/features/sqlRunner/store/tableVisSlice.ts

```diff
@@ -1,19 +1,24 @@
 import {
     ChartKind,
+    deepEqual,
     isTableChartSQLConfig,
     type TableChartSqlConfig,
 } from '@lightdash/common';
 import type { PayloadAction } from '@reduxjs/toolkit';
 import { createSlice } from '@reduxjs/toolkit';
 import { setSavedChartData, setSqlRunnerResults } from './sqlRunnerSlice';
 
-const initialState: { config: TableChartSqlConfig | undefined } = {
-    config: undefined,
+type InitialState = {
+    defaultColumnConfig: TableChartSqlConfig['columns'] | undefined;
+    config: TableChartSqlConfig | undefined;
 };
 
 export const tableVisSlice = createSlice({
     name: 'tableVisConfig',
-    initialState,
+    initialState: {
+        defaultColumnConfig: undefined,
+        config: undefined,
+    } as InitialState,
```

Comment:
do we need this? 🤔

Reply:
Probably not.

Reply:
Actually yes, there are ways to avoid it but none is clearer or less work than this.